sourcegraph/sourcegraph
# cloud: guard o11y for non-operator users

This is a followup of https://github.com/sourcegraph/sourcegraph/pull/45827, which further protects the debug endpoints in the backend from non-Sourcegraph operators, not just hiding from the UI.

Part of

- https://github.com/sourcegraph/customer/issues/1743
- https://github.com/sourcegraph/security-issues/issues/330

## Test plan

1. Get the `cloud-site-config.json` (or use the `signed-clout-site-config` from [1password](https://start.1password.com/open/i?a=HEDEDSLHPBFGRBTKAKJWE23XX4&v=dnrhbauihkhjs5ag6vszsme45a&i=eyhez5jvjmdd745zu6jwferxha&h=team-sourcegraph.1password.com)
2. Get the signed config
   ```sh
   mi sign-cloud-site-config --config-file ./cloud-site-config.json
   ```
3. Add the following to `sg.config.overwrite.yaml`
   ```yaml
   commands:
     frontend:
       env:
         SRC_CLOUD_SITE_CONFIG: "REDACTED"
   ```
4. `DEV_WEB_BUILDER=esbuild sg start`
5. Create `sourcegraph-operator-site-admin-hide-maintenance` feature flag and set to `true`
6. Trying to directly access the `https://sourcegraph.test:3443/-/debug/` should fail:
	<img width="412" alt="CleanShot 2023-04-28 at 15 49 13@2x" src="https://user-images.githubusercontent.com/2946214/235239892-8acb7ea2-134c-48e4-b369-c0ee5a5d79bd.png">
7. Sign in with SOAP should load the page as normal
	<img width="468" alt="CleanShot 2023-04-28 at 15 51 32@2x" src="https://user-images.githubusercontent.com/2946214/235240259-864fc6e1-73bd-4972-85a9-6d39a498eec1.png">
